File: Number_Theory/020 Check square number.py
'''
    Ex20: Check Square number (1<= N <= 1e18)
    Status: TLE, accepted with C++
'''

import math

def check_Square(n):
    i = 0
    while i*i <= n:
        if i*i == n:
            return True
        i = i+1
    return False

n = int(input())

print("YES") if check_Square(n) else print("NO")

File: Number_Theory/020 Check square number.cpp
/*
    Ex20:   Check Square number (1<= N <= 1e18)
    Status: acp
*/
#include <iostream>
#include <cmath>
#include <algorithm>
#include<vector>
#define ll long long 
using namespace std ;

bool check(ll n)
{
    for(ll i = 0; i*i<=n;i++)
    {
        if(i*i == n)
            return true;
    }
    return false;
}

int main()
{
    ll n; cin >> n;
    if(check(n)) cout << "YES";
    else cout << "NO";
    return 0;
}